airbyte-scheduler/src/main/java/io/airbyte/scheduler/temporal/CheckConnectionWorkflow.java

```diff
@@ -0,0 +1,127 @@
+
+@WorkflowInterface
+public interface CheckConnectionWorkflow {
+
+  @WorkflowMethod
+  JobOutput run(IntegrationLauncherConfig launcherConfig, StandardCheckConnectionInput connectionConfiguration) throws TemporalJobException;
+
+  class WorkflowImpl implements CheckConnectionWorkflow {
+
+    ActivityOptions options = ActivityOptions.newBuilder()
+        .setScheduleToCloseTimeout(Duration.ofMinutes(2)) // todo
+        .build();
+
+    private final CheckConnectionActivity activity = Workflow.newActivityStub(CheckConnectionActivity.class, options);
+
+    @Override
+    public JobOutput run(IntegrationLauncherConfig launcherConfig, StandardCheckConnectionInput connectionConfiguration) throws TemporalJobException {
+      return activity.run(launcherConfig, connectionConfiguration);
+    }
+
+  }
+
+  @ActivityInterface
+  interface CheckConnectionActivity {
+
+    @ActivityMethod
+    JobOutput run(IntegrationLauncherConfig launcherConfig, StandardCheckConnectionInput connectionConfiguration) throws TemporalJobException;
+
+  }
+
+  class CheckConnectionActivityImpl implements CheckConnectionActivity {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CheckConnectionActivityImpl.class);
+
+    private final ProcessBuilderFactory pbf;
+    private final Path workspaceRoot;
+
+    public CheckConnectionActivityImpl(ProcessBuilderFactory pbf, Path workspaceRoot) {
+      this.pbf = pbf;
+      this.workspaceRoot = workspaceRoot;
+    }
+
+    public JobOutput run(IntegrationLauncherConfig launcherConfig, StandardCheckConnectionInput connectionConfiguration) throws TemporalJobException {
+      try {
+        // todo (cgardens) - there are 2 sources of truth for job path. we need to reduce this down to one,
+        // once we are fully on temporal.
+        final Path jobRoot = workspaceRoot
+            .resolve(String.valueOf(launcherConfig.getJobId()))
+            .resolve(String.valueOf(launcherConfig.getAttemptId().intValue()));
+
+        MDC.put("job_id", String.valueOf(launcherConfig.getJobId()));
+        MDC.put("job_root", jobRoot.toString());
+        MDC.put("job_log_filename", WorkerConstants.LOG_FILENAME);
+
+        final IntegrationLauncher integrationLauncher =
+            new AirbyteIntegrationLauncher(launcherConfig.getJobId(), launcherConfig.getAttemptId().intValue(), launcherConfig.getDockerImage(), pbf);
+        final AirbyteStreamFactory streamFactory = new DefaultAirbyteStreamFactory();
+        final OutputAndStatus<JobOutput> run =
+            new JobOutputCheckConnectionWorker(new DefaultCheckConnectionWorker(integrationLauncher, streamFactory)).run(connectionConfiguration,
+                jobRoot);
+        if (run.getStatus() == JobStatus.SUCCEEDED) {
+          Preconditions.checkState(run.getOutput().isPresent());
+          LOGGER.info("job output {}", run.getOutput().get());
+          return run.getOutput().get();
+        } else {
+          throw new TemporalJobException();
```

Comment:
I don't think this is going to work. You're catching and re-throwing as a runtime